emr-launcher.tf

```diff
@@ -0,0 +1,102 @@
+variable "adg_emr_launcher_zip" {
+  type = map(string)
+
+  default = {
+    base_path = ""
+    version   = ""
+  }
+}
+
+resource "aws_lambda_function" "adg_emr_launcher" {
+  filename      = "${var.adg_emr_launcher_zip["base_path"]}/emr-launcher-${var.adg_emr_launcher_zip["version"]}.zip"
+  function_name = "adg_emr_launcher"
+  role          = aws_iam_role.adg_emr_launcher_lambda_role.arn
+  handler       = "emr_launcher.handler"
+  runtime       = "python3.7"
+  source_code_hash = filebase64sha256(
+  format(
+  "%s/emr-launcher-%s.zip",
+  var.adg_emr_launcher_zip["base_path"],
+  var.adg_emr_launcher_zip["version"]
+  )
+  )
+  publish = false
+  timeout = 60
+
+  environment {
+    variables = {
+      EMR_LAUNCHER_CONFIG_DIR = "docs/examples"
+      EMR_LAUNCHER_LOG_LEVEL = "debug"
+    }
+  }
+}
+
+resource "aws_iam_role" "adg_emr_launcher_lambda_role" {
+  name               = "adg_emr_launcher_lambda_role"
+  assume_role_policy = data.aws_iam_policy_document.adg_emr_launcher_assume_policy.json
+}
+
+data "aws_iam_policy_document" "adg_emr_launcher_assume_policy" {
+  statement {
+    sid    = "AdgEmrLauncherLambdaAssumeRolePolicy"
+    effect = "Allow"
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      identifiers = ["lambda.amazonaws.com"]
+      type = "Service"
+    }
+  }
+}
+
+data "aws_iam_policy_document" "adg_emr_launcher_read_secret_policy" {
```

Comment:
Can you split this policy in 2 please?

The secretsmanager one only needs access to GetSecretValue on the ADG-specific secret
The emr one only needs access to RunJobFlow, but will need it and won't need (i.e. can't take) a resource constraint.

Reply:
Fixed by edad3fb